jax/nn/initializers.py

```diff
@@ -96,3 +97,31 @@ def init(key, shape, dtype=np.float32):
     Q = np.moveaxis(Q, -1, column_axis)
     return scale * Q
   return init
+
+
+def delta_orthogonal(scale=1.0, column_axis=-1):
+  """
+  Construct an initializer for delta orthogonal kernels; see arXiv:1806.05393. 
+
+  The shape must be 3D, 4D or 5D.
+  """
+  def init(key, shape, dtype=np.float32):
+    if len(shape) not in [3, 4, 5]:
+      raise ValueError("Delta orthogonal initializer requires a 3D, 4D or 5D "
+                       "shape.")
+    if shape[-1] < shape[-2]:
+      raise ValueError("`fan_in` must be less or equal than `fan_out`. ")
+    ortho_init = orthogonal(scale=scale, column_axis=column_axis)
+    ortho_matrix = ortho_init(key, shape[-2:])
+    W = np.zeros(shape)
+    if len(shape) == 3:
+      k = shape[0]
+      return ops.index_update(W, [(k-1)//2, ...], ortho_matrix)
```

Comment:
This syntax doesn't work on Python 2 (and causes CI to fail, as you pointed out). Our `index_update` function is intended to accept any NumPy indexer/sequence of indicators, but a list isn't a valid syntactic environment for e.g. `2:4` or (on Python 2) `...`. In order to perform index_update indexing with indexers that aren't allowed outside an indexing environment, use `ops.index`:
```suggestion
      return ops.index_update(W, ops.index[(k-1)//2, ...], ortho_matrix)
```
(and similarly for the other two cases).